contracts/GradualDeliveryCrowdsale.sol

```diff
@@ -96,4 +98,59 @@ contract GradualDeliveryCrowdsale is Crowdsale, Ownable {
         beneficiaries.push(_beneficiary);
         balances[_beneficiary] = balances[_beneficiary].add(_tokenAmount);
     }
+    /**
+     * @dev Refund the given ether to a beneficiary.  It only can be called by
+     * either the contract owner or the wallet (i.e. Crowdsale.wallet) address.
+     * The only amount of the ether sent together in a transaction is refunded.
+     */
+    function depositRefund(address _beneficiary) public payable {
+        require(msg.sender == owner || msg.sender == wallet);
+        uint256 weiToRefund = msg.value;
+        require(weiToRefund <= weiRaised);
+        uint256 tokensToRefund = _getTokenAmount(weiToRefund);
+        uint256 tokenBalance = balances[_beneficiary];
+        require(tokenBalance >= tokensToRefund);
+        weiRaised = weiRaised.sub(weiToRefund);
+        balances[_beneficiary] = tokenBalance.sub(tokensToRefund);
+        refundedDeposits[_beneficiary] = weiToRefund;
```

Comment:
Yes, according to [Solidity's docs about mapping types][1]:

> Mappings can be seen as [hash tables][2] which are virtually initialized such that every possible key exists and is mapped to a value whose byte-representation is all zeros: a type's [default value][3]. The similarity ends here, though: The key data is not actually stored in a mapping, only its `keccak256` hash used to look up the value.
>
> Because of this, mappings do not have a length or a concept of a key or value being “set”.

This property makes Solidity's mappings differ from dictionaries/maps in other programming languages, and treating them as a kind of [`defaultdict`][4] is norm in smart contracts.

[1]: https://solidity.readthedocs.io/en/v0.4.21/types.html#mappings
[2]: https://en.wikipedia.org/wiki/Hash_table
[3]: https://solidity.readthedocs.io/en/v0.4.21/control-structures.html#default-value
[4]: https://docs.python.org/3/library/collections.html#collections.defaultdict

Reply:
@dahlia maybe my comment was unclear so let me rephrase... It looks like if this function runs twice the first `refundedDeposits[_beneficiary]` is overwritten. Is that okay?

Reply:
Seems a valid point.  I'll fix it.

Reply:
@qria Fixed in ce894a57f14ab082249085eafe87f0728c1fe1ff.